eddiejaoude/stargate
# feat: initial bot code

#### Proposed Changes

This PR establishes some initial bot code. The changes include:
- A bot configuration object (to allow passing values between commands, such as the prefix)
- A defined interface for command files
- Manual compiling of commands into an array
- A `help` and `ping` command, for testing purposes

After merge, still to do:
- Implement error handling for failed commands, login, etc.
- Migrate the manual command list to an automatic function

Steps to test these changes:
- Pull down changes locally
- Add your bot token to the `.env` (visit the [Discord Developer Portal]() to create an application if you do not have one)
- Install dependencies with `npm ci`
- Build the TypeScript files with `npm run build`
- Start the application with `npm start`
  - Confirm you see "It's aliiiiiive" in the terminal - this means the bot logged in
- Visit a Discord channel where your bot application is present
  - Send the message "!stargate ping" and see the bot's response
  - Send the message "!stargate help" and see the bot's response
  - Send the message "!stargate help help" and see the bot's response
  - Send the message "!stargate hahaha" and see no response from the bot
 

#### Screenshots
